### PR TITLE
Removed calculateDistance from exported methods

### PR DESCRIPTION
**Pull request by TuomasBorman**
Hi,

this PR removes `calculateDistance` from exported methods. 

Issue: https://github.com/microbiome/mia/issues/77

-Tuomas

### CODE REVIEW

**Review by antagomir — CHANGES_REQUESTED** [review]
Otherwise ok but include [.Deprecated](https://stat.ethz.ch/R-manual/R-devel/library/base/html/Deprecated.html) message within the function.

File: DESCRIPTION [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Package: mia
 Type: Package
-Version: 1.1.9
+Version: 1.2.0
```
Comment (FelixErnst) [inline DESCRIPTION]: Thats wrong.  `1.1.10`

**Review by FelixErnst — CHANGES_REQUESTED** [review]
Please have a look at https://bioconductor.org/developers/how-to/deprecation/

The function needs to be available, but showing a deprecated message using `.Deprected` as Leo has mentioned.

Since no serious S4 dispatch was used for `calculateDistance`, just put the code into a new internal function named `.calculate_distance` and call that internal wrapper instead of `calculateDistance`. With next devel just remove the `setMethod` part and you are done.

File: R/calculateDistance.R [hunk 2]
```diff
@@ -46,7 +50,8 @@ setGeneric("calculateDistance", signature = c("x"),
 #' @export
 setMethod("calculateDistance", signature = c(x = "ANY"),
     function(x, FUN = stats::dist, ...){
-        do.call(FUN, c(list(x),list(...)))
+        .Deprecated("directly 'FUN' argument's specified function")
```
Comment (FelixErnst) [inline R/calculateDistance.R]: What is the output of this? The message in itself seems to be a bit hard to understand. Can you make the message more verbose?

Comment (TuomasBorman) [reply R/calculateDistance.R]: It is 

```
> calculateDistance(se)
         Sample1  Sample2  Sample3  Sample4  Sample5
Sample2 4.242641                                    
Sample3 5.385165 1.732051                           
Sample4 5.385165 4.358899 4.000000                  
Sample5 3.464102 3.162278 3.605551 2.236068         
Sample6 4.123106 1.000000 2.000000 4.242641 3.000000
Warning message:
'calculateDistance' is deprecated.
Use 'directly 'FUN' argument's specified function' instead.
See help("Deprecated") 
```

The problem is that `FUN` contains the function, not function name, so I'm not sure what is good way

Comment (TuomasBorman) [reply R/calculateDistance.R]: It is also possible to create own message. I think its better so we get rid off those extra quotation marks
```

> calculateDistance(se)
         Sample1  Sample2  Sample3  Sample4  Sample5
Sample2 4.242641                                    
Sample3 5.385165 1.732051                           
Sample4 5.385165 4.358899 4.000000                  
Sample5 3.464102 3.162278 3.605551 2.236068         
Sample6 4.123106 1.000000 2.000000 4.242641 3.000000
Warning message:
This is also possible... 
```

Comment (TuomasBorman) [reply R/calculateDistance.R]: Current warning message

```
> calculateDistance(se)
         Sample1  Sample2  Sample3  Sample4  Sample5
Sample2 4.242641                                    
Sample3 5.385165 1.732051                           
Sample4 5.385165 4.358899 4.000000                  
Sample5 3.464102 3.162278 3.605551 2.236068         
Sample6 4.123106 1.000000 2.000000 4.242641 3.000000
Warning message:
'calculateDistance' is deprecated. 
Instead, use directly the function that is specified by 'FUN' argument. 
See help('Deprecated') 
```

**Review by FelixErnst — APPROVED** [review]

**Pull request merged**